utils/obsproc/Rads2Ioda.h

```diff
@@ -13,75 +13,49 @@
 
 #include "NetCDFToIodaConverter.h"
 
-#include "oops/util/missingValues.h"
-
 namespace gdasapp {
+
   class Rads2Ioda : public NetCDFToIodaConverter {
    public:
     explicit Rads2Ioda(const eckit::Configuration & fullConfig)
     : NetCDFToIodaConverter(fullConfig) {
-    oops::Log::info() << "Window begin: " << this->windowBegin_ << std::endl;
-    oops::Log::info() << "Window end: " << this->windowEnd_ << std::endl;
-    oops::Log::info() << "IODA output file name: " << this->outputFilename_ << std::endl;
     }
 
-    // Read netcdf file and populate group
-    void ReadFromNetCDF(ioda::Group & group) override {
-      std::string fileName = this->inputFilenames_[0];  // TODO(Guillaume): make it work for a list
+    // Read netcdf file and populate iodaVars
+    void ProviderToIodaVars(std::string fileName, gdasapp::IodaVars & iodaVars) override {
+      oops::Log::info() << "Processing files provided by the RADS" << std::endl;
+
+      // Set nVars to 1
+      iodaVars.nVars = 1;
 
       // Open the NetCDF file in read-only mode
       netCDF::NcFile ncFile(fileName, netCDF::NcFile::read);
 
       // Get dimensions
-      int location = ncFile.getDim("time").getSize();
-      int nVars = 1;
+      iodaVars.location = ncFile.getDim("time").getSize();
+      oops::Log::debug() << "--- iodaVars.location: " << iodaVars.location << std::endl;
+
+      // Allocate memory
+      iodaVars.obsVal = new Eigen::ArrayXf(iodaVars.location);
+      iodaVars.obsError = new Eigen::ArrayXf(iodaVars.location);
+      iodaVars.preQc = new Eigen::ArrayXi(iodaVars.location);
```

Comment:
No need for `new` if vars are value refs.

Reply:
:+1: